### ISSUE TITLE
jesec-libtorrent failing test

### ISSUE THREAD

**Issue by NULLx76** [NixOS/nixpkgs#167854]
### Describe the bug
jesec-libtorrent tests fail, making the build fail.

### Steps To Reproduce
1. build jesec-libtorrent
2. test fails, failing the build

### Expected behavior
I expect it to build succesfully

### Screenshots
![image](https://user-images.githubusercontent.com/1809198/162423769-e3a5262d-ab98-4ca0-83b4-53d3aa9480b9.png)

![image](https://user-images.githubusercontent.com/1809198/162423796-01b95408-e1fa-4167-881b-60b8c988cc54.png)

### Additional context
I'm building the package with the help of morph to deploy on a server, though this shouldn't really have an impact.

### Notify maintainers
@winterqt @AndersonTorres 

### Metadata
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.174-2-pve, NixOS, 22.05 (Quokka), 22.05pre-git`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.0pre20220322_d5d4d98`
 - channels(root): `"nixos-20.09.3246.d4189f68fdb"`
 - nixpkgs: `/home/victor/src/nixos-configs/nix-channel/nixpkgs`
```

**Comment by winterqt** [comment]
This happened between 2d7c764b78c8038fef0256c48d9255d795645dc6 and f22dc92e9ff0444f1dae325ca30bbdc67972a8cf.

I'll bisect later, thanks for the heads up!

**Comment by winterqt** [comment]
The test is expecting `EAI_NONAME` to be returned by `getaddrinfo("1.1.1.300")`, but is instead getting `EAI_AGAIN`.

I'm... really not sure how to move forward with this, short of disabling the test.

**Comment by AndersonTorres** [comment]
Next time copy-paste the error message.

**Comment by vcunat** [comment]
Perhaps triggered by upgrading glibc from 2.33 to 2.34?  (some change in getaddrinfo() behavior)

**Comment by winterqt** [comment]
I ruled that out initially since I didn't see anything mentioned in the changelog about getaddrinfo. I'll give it a try, though.

**Comment by winterqt** [comment]
This has been fixed by #169236.
